rmgpy/rmg/main.py

```diff
@@ -1463,21 +1481,126 @@ def generateCanteraFiles(self, chemkinFile, **kwargs):
+                    for j, core_spcs in enumerate(self.reactionModel.core.species):
+                        if core_spcs.isIsomorphic(spcs):
+                            core_species_map[index] = core_spcs
+                            break
+                    else:
+                        raise RuntimeError('Species {0} was defined in the restart file, but was not included in the'
+                                           'core.'.format(spcs))
+                    self.reactionModel.core.species.pop(j)
+
+                remaining_species = self.reactionModel.core.species
+
+                reordered_core_species = [core_species_map[i] for i in range(len(core_species_map))] + remaining_species
+                self.reactionModel.core.species = reordered_core_species
+
+                # If we are restarting we must be able to handle all four possible combinations of whether or not
+                # filtering was used in the restart job or if filtering is used in the current job. To summarize:
+                #
+                # If filtering is being used in this job, we only need to set the threshold flags properly because the
+                # react flags are updated based on this before the first enlarge. Otherwise, we need to explicitly set
+                # the react flags properly because the thresholds will not be used.
+                #
+                # If filtering was used in the job we are restarting from, we can load these values in for all of the
+                # species present in that job to help us set the current threshold or react flags.
+                # Otherwise, if filtering was not used in the previous job then we can assume that all species
+                # present in the job we are restarting from have already reacted, and thus either set their
+                # filters to be True or their react flags to be false.
+
+                # Fill in the filter tensors.
+                if filters_found:  # Add in the filter data where we have it.
+                    # Note that additional species that have been defined in the input file but were not
+                    # present in the restart core seed will be reacted.
+
+                    if self.filterReactions:  # Filling in the filter thresholds will suffice
+                        # Fill in the newly initialized filter tensors
+                        self.unimolecularThreshold = unimolecularThreshold
+                        self.bimolecularThreshold = bimolecularThreshold
+                        if self.trimolecular:
+                            self.trimolecularThreshold = trimolecularThreshold
+
+                    else:  # We must set the react flags instead. If it was `True` in the threshold, it should not react
+                        self.unimolecularReact = np.logical_not(unimolecularThreshold)
+                        self.bimolecularReact = np.logical_not(bimolecularThreshold)
```

Comment:
Is `bimolecularThreshold` triangular or the full matrix? Is there any danger in inverting the entire thing if it is triangular?

Reply:
It would be triangular because it is saved from RMG. In theory this could cause problems if we ever read from the lower triangle, but this is true regardless of the changes implemented in this PR.